platform/entanglement/src/services/CopyService.js

```diff
@@ -54,45 +57,156 @@ define(
+             * the farthest leaf, then works its way back up again,
+             * cloning objects, and composing them with their child clones
+             * as it goes
+             * @param originalObject
+             * @param originalParent
+             * @returns {*}
+             */
+            function copy(originalObject, originalParent) {
+                //Make a clone of the model of the object to be copied
+                var modelClone = {
+                        id: uuid(),
+                        model: makeClone(originalObject.getModel()),
+                        persistenceSpace: originalParent.getCapability('persistence')
+                    };
+                delete modelClone.model.composition;
+                delete modelClone.model.persisted;
+                delete modelClone.model.modified;
+                return $q.when(originalObject.useCapability('composition')).then(function(composees){
+                    progress({phase: "preparing"});
+                    return (composees || []).reduce(function(promise, composee){
+                            //If the object is composed of other
+                            // objects, chain a promise..
+                            return promise.then(function(){
+                                // ...to recursively copy it (and its children)
+                                return copy(composee, originalObject).then(function(composeeClone){
+                                    //Once copied, associate each cloned
+                                    // composee with its parent clone
+                                    if ( !(composee.hasCapability("location") && composee.getCapability("location").isLink())) {
+                                        //If the object is not a link,
+                                        // locate it within its parent
+                                        composeeClone.model.location = modelClone.id;
+                                    }
+                                    modelClone.model.composition = modelClone.model.composition || [];
+                                    return modelClone.model.composition.push(composeeClone.id);
+                                });
+                            });}, $q.when(undefined)
+                    ).then(function (){
+                        //Add the clone to the list of clones that will
+                        //be returned by this function
+                        clones.push(modelClone);
+                        return modelClone;
+                    });
+                });
             }
+            
+            return copy(domainObject, parent).then(function(domainObjectClone){
+                //If the domain object being cloned is not a link, set its
+                // location to the new parent
+                if ( !(domainObject.hasCapability("location") && domainObject.getCapability("location").isLink())) {
```

Comment:
That is correct, links become originals currently. This is counter-intuitive to me as well, but during design sessions this was what users indicated they would expect. We may end up refining that after feedback (Move/Copy/Link hasn't had much actual user time)

The only case where we'd want to use links instead of creating new originals is if objects are not creatable (in that case, links should be made.) This has not been implemented, though, and I'd consider that out of scope for this issue.